idl.md

```diff
@@ -1,61 +1,90 @@
-## Proposed IDL
-```javascript
+# Proposed IDL
+
+```js
 partial interface ServiceWorkerRegistration {
-  readonly attribute SyncManager syncManager;
+  readonly attribute SyncManager sync;
+  readonly attribute PeriodicSyncManager periodicSync;
 };
 
 interface SyncManager {
   Promise<SyncRegistration> register(optional SyncRegistrationOptions options);
-  Promise<SyncRegistration> getRegistration(DOMString id);
+  Promise<SyncRegistration> getRegistration(DOMString tag);
   Promise<sequence<SyncRegistration>> getRegistrations();
-  Promise<SyncPermissionStatus> hasPermission();
-  readonly attribute unsigned long minAllowablePeriod;
+  Promise<SyncPermissionState> permissionState();
```

Comment:
Re: get vs. getRegistration, getAll vs. getRegistrations, the current naming seems consistent with the ServiceWorker spec. I also think that consistency is a good reason to stick with the established convention. 

https://slightlyoff.github.io/ServiceWorker/spec/service_worker/#service-worker-container

I think it can also avoid confusion about what to expect by *getting* from nav.sync (i.e. not something you can ask to "sync, presto!" but a representation of your registered intent to be able to do bg sync work).  WYGIWYAF: what you get is what you ask for?


Re: permissionState(), SyncPermissionState are consistent with the Permissions API but not with Push's hasPermission() and PushPermissionStatus, and it feels a bit odd that you can *get* a registration but can't *get* a permissionState  :(